### ISSUE TITLE
make toolbox search compatible with keyboard navigation

### ISSUE THREAD

**Issue by maribethb** [rachel-fenichel/blockly-samples#2613]
### Check for duplicates

- [x] I have searched for similar issues before opening a new one.

### Component

toolbox-search

### Request

Toolbox search does not work with keyboard navigation currently.

When you use the arrow keys to navigate to the search category, the input box has focus. There is no way to use keyboard navigation to then move to the flyout that has the search results. If you press escape, currently the search is cleared. There is also no way to use the arrow keys to move to a different toolbox category once the search has focus.

To solve this, we'll likely need a 2 step focus system, where you can be focused on the toolbox category separately from the search input bar, so that you can navigate separately from the text input. This would be similar to how we have things set up for comments.
